config/v1alpha1/types_cluster_monitoring.go

```diff
@@ -416,6 +440,432 @@ type MetricsServerConfig struct {
 	TopologySpreadConstraints []v1.TopologySpreadConstraint `json:"topologySpreadConstraints,omitempty"`
 }
 
+// PrometheusK8sConfig provides configuration options for the Prometheus instance
+// Use this configuration to control
+// Prometheus deployment, pod scheduling, resource allocation, retention policies, and external integrations.
+// +kubebuilder:validation:MinProperties=1
+type PrometheusK8sConfig struct {
+	// additionalAlertmanagerConfigs configures additional Alertmanager instances that receive alerts from
+	// the Prometheus component. This is useful for organizations that need to:
+	//   - Send alerts to external monitoring systems (like PagerDuty, Slack, or custom webhooks)
+	//   - Route different types of alerts to different teams or systems
+	//   - Integrate with existing enterprise alerting infrastructure
+	//   - Maintain separate alert routing for compliance or organizational requirements
+	// By default, no additional Alertmanager instances are configured.
+	// Maximum of 10 additional Alertmanager configurations can be specified.
+	// When omitted, no additional Alertmanager instances are configured (default behavior).
+	// When set to an empty array [], the behavior is the same as omitting the field.
+	// +optional
+	// +kubebuilder:validation:MaxItems=10
+	// +listType=atomic
+	AdditionalAlertmanagerConfigs []AdditionalAlertmanagerConfig `json:"additionalAlertmanagerConfigs,omitempty"`
+	// enforcedBodySizeLimit enforces a body size limit for Prometheus scraped metrics. If a scraped
+	// target's body response is larger than the limit, the scrape will fail.
+	// The following values are valid:
+	// a numeric value in Prometheus size format (such as "4MB", "1000", "1GB", "512KB", "100B")
+	// or the string `automatic`, which indicates that the limit will be
+	// automatically calculated based on cluster capacity.
+	// To specify no limit, omit this field.
+	// The value must match the following pattern: ^(automatic|[0-9]+(B|KB|MB|GB|TB)?)$
+	// Minimum length is 1 character.
+	// Maximum length is 50 characters.
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=50
+	// +optional
+	EnforcedBodySizeLimit string `json:"enforcedBodySizeLimit,omitempty"`
+	// externalLabels defines labels to be added to any time series or alerts when
+	// communicating with external systems such as federation, remote storage,
+	// and Alertmanager. By default, no labels are added.
+	// When omitted, no external labels are applied (default behavior).
```

Comment:
Nit: I think that just saying this is sufficient:
```suggestion
	// and Alertmanager.
	// When omitted, no external labels are applied.
```